templates/tapestry.js

```diff
@@ -1374,329 +1384,28 @@ function tapestryTool(config){
     /****************************************************
      * MEDIA RELATED FUNCTIONS
      ****************************************************/
-    
-    function setupLightbox(id, mediaFormat, mediaType, mediaUrl, width, height) {
-        // Adjust the width and height here before passing it into setup media
-        var lightboxDimensions = getLightboxDimensions(height, width);
-        width = lightboxDimensions.width;
-        height = lightboxDimensions.height;
-        var media = setupMedia(id, mediaFormat, mediaType, mediaUrl, width, height);
-    
-    $('<div id="spotlight-overlay"></div>').on("click", function(){
-            closeLightbox(id, mediaType);
-            exitViewMode();
-        }).appendTo('body');
-    
-        var top = lightboxDimensions.adjustedOn === "width" ? ((getBrowserHeight() - height) / 2) + $(this).scrollTop() : (NORMAL_RADIUS * 1.5) + (NORMAL_RADIUS * 0.1);
-        $('<div id="spotlight-content" data-view-mode="' + (enablePopupNodes ? 'true' : 'false') + '" data-media-format="' + mediaFormat + '" data-media-type="' + mediaType + '"><\/div>').css({
-            top: top,
-            left: (getBrowserWidth() - width) / 2,
-            width: width,
-            height: height,
-            opacity: 0
-        }).appendTo('body');
-    
-        // We don't want draggable for text because we want the user to be able to select text
-        if (mediaType != "text") {
-            $('#spotlight-content').draggable({
-                delay: 10,
-                distance: 8
-            });
-        }
-    
-    $("<div class='media-wrapper'></div>").appendTo('#spotlight-content');
-    media.appendTo('.media-wrapper');
-    
-    $("<button id='lightbox-close-wrapper'><div class='lightbox-close'><i class='fa fa-times'</i></div></button>")
-            .on("click", function() {
-                closeLightbox(id, mediaType);
-                exitViewMode();
-            })
-            .appendTo('#spotlight-content');
-    
-        setTimeout(function(){
-            $('#spotlight-content').css({
-                opacity: 1
-            });
-            if (mediaType != 'video') {
-                updateMediaIcon(id, mediaType);
-            }
-        }, 1000);
-    
-        if (mediaType === "video") {
-            var loadEvent = 'load';
-            if (mediaFormat == "mp4") {
-                loadEvent = "loadstart";
+
+    this.updateChildren = function(id, video) {
```

Comment:
Please leave a comment to explain what this function does. Also, if it's possible, I prefer to have functions in their own components rather than in this file as much as possible. We want to eventually get rid of this altogether, so if you could move this function into the component that uses it, it would be better (potentially put into a service and consume it in the component). You can always `setDataset()` from the component to push changes to `tapestry.js`.

Reply:
I looked into this a little bit more and to me I feel like I would keep it there for now at least until this file becomes its own component. For now I added a comment that describes what this function does.

While this function **is** only used by the `VideoMedia` component, it does manipulate the entire Tapestry. So we have 2 options, which is to pass in the entire tapestry to `VideoMedia`, or move the functionality to `Tapestry.vue`. I feel that passing the entire tapestry is wasteful, and `Tapestry.vue` is bloated as it is.

But let me know what you think.

Reply:
Ok, I see your point. We can wait until this becomes a component to make that change. Thanks